### DISCUSSION TITLE
It looks like setError is not working correctly in newest versions

### DISCUSSION THREAD

**Discussion by superpuper777** [react-hook-form/react-hook-form#7992] [Q&A]
In useForm/setError section (https://react-hook-form.com/api/useform/seterror), I open multiple errors example on sandbox (https://codesandbox.io/s/react-hook-form-v7-seterror-3y1op). When I click on "Trigger Name Errors" button and then submit button, form works correctly (react-hook-from version 7.0.0-beta.13). But when I choose latest react-hook-form version (7.27.1) and repeat previous steps, form submits empty data. What should be the behavior? 
Thank you

**Comment by bluebill1049** [comment]
thanks for pointing that out, I have replaced it with `setValue` API instead of inline `onChange` without given `name`.

**Reply by superpuper777** [reply]
After form submit, when we use setValue like in new example, errors reset, is it right behavior?

**Reply by bluebill1049** [reply]
because the source of truth for validation lives with `register`, setting an error will trigger state update to flush down errors, while rerun all the validation it will pass, if this is not desired behavior attach validation rules alone with register and trigger the validation.
